### PR TITLE
Support OPERATE for FieldData with global storage.

### PR DESCRIPTION
**Pull request by blattms**
Extracted from #4181 with added tests.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/FieldProps.hpp [hunk 1]
```diff
@@ -704,6 +704,9 @@ class FieldProps {
     void deleteMINPVV();
 
 private:
+    template<typename T>
+    static void update_global_from_local(Fieldprops::FieldData<T>& data,
+                                         const std::vector<Box::cell_index>& index_list);
```
Comment (bska) [inline opm/input/eclipse/EclipseState/Grid/FieldProps.hpp:709]: Why does this need to be a static member of the class?  It doesn't access any of the members of the class and could just as well be implemented as a regular function template in `FieldProps.cpp`.

**Review by bska — COMMENTED** [review]
Looks good for the most part.  The only thing I think I'd structure differently is the `update_global_from_local()` helper function which does not need to be a member of the class.  Just make it a normal function template in `FieldProps.cpp`'s anonymous namespace&ndash;e.g., next to the [`apply()`](https://github.com/OPM/opm-common/blob/4356fa9034f044baa538aad986814c7698fee738/opm/input/eclipse/EclipseState/Grid/FieldProps.cpp#L511) function.

**Comment by blattms** [comment]
Thanks, I'll make it  a free function then.

**Comment by blattms** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and as the build check is green I'll merge into master.

**Pull request merged**